libethereum/EthereumHost.h

```diff
@@ -132,7 +132,6 @@ class EthereumHost: public p2p::HostCapability<EthereumPeer>, Worker
 	bool m_newTransactions = false;
 	bool m_newBlocks = false;
 
-	mutable RecursiveMutex x_sync;
```

Comment:
I have some doubts about removing this mutes. Some operations that were atomic now are not atomic...

At least let's add `RecursiveGuard l(x_sync);` inside `BlockChainSync::competeSync()` and `BlockChainSync::abortSync()`, otherwise they now stay unprotected

Reply:
Yes, I think I should do the "at least" plan.

Reply:
Oh, those `RecursiveGuard`s are already added on `develop`.